Plugins/Pack/Pack/ALAR.cs

```diff
@@ -90,5 +113,125 @@ public static sFolder Unpack_Type2(sFile file)
             br.Close();
             return unpacked;
         }
+
+        public sFolder Unpack_Type3(sFile file)
+        {
+            BinaryReader br = new BinaryReader(File.OpenRead(file.path));
+            sFolder unpacked = new sFolder();
+            unpacked.files = new List<sFile>();
+            unpacked.folders = new List<sFolder>();
+
+            // Read the header file
+            char[] header = br.ReadChars(4);
+            byte type = br.ReadByte();
+            byte unk = br.ReadByte();
+            uint num_files = br.ReadUInt32();
+            ushort unk2 = br.ReadUInt16();
+            uint array_count = br.ReadUInt32();
+            ushort endFileIndex = br.ReadUInt16();
+            ushort[] fileTableIndex = new ushort[array_count + 1];
+
+            for (int i = 0; i < array_count + 1;i++)
+            {
+                fileTableIndex[i] = br.ReadUInt16();
+            }
+
+            // Index table
+            foreach(ushort filePosition in fileTableIndex)
+            {
+                br.BaseStream.Position = filePosition;
+
+                ushort fileID = br.ReadUInt16();
+                ushort unk3 = br.ReadUInt16();
+
+                sFile cfile = new sFile
+                {
+                    path = file.path,
+                    offset = br.ReadUInt32(), // StartOfFile
+                    size = br.ReadUInt32() // SizeOfFile
+                };
+
+                ushort unk4 = br.ReadUInt16();
+                ushort unk5 = br.ReadUInt16();
+                ushort unk6 = br.ReadUInt16();
+
+                string filename = this.ReadNullTerminatedString(br);
+
+                AddFile(unpacked, cfile, filename);
+            }
+
+            br.Close();
+            return unpacked;
+        }
+
+        public sFolder Decompress_LZSS(sFile file)
+        {
+            sFolder unpacked = new sFolder();
+            unpacked.files = new List<sFile>();
+
+            string temp = pluginHost.Get_TempFolder() + Path.DirectorySeparatorChar + Path.GetRandomFileName();
+            Byte[] compressFile = new Byte[(new FileInfo(file.path).Length) - 4];
+            Array.Copy(File.ReadAllBytes(file.path), 4, compressFile, 0, compressFile.Length);
+            File.WriteAllBytes(temp, compressFile);
+
+            // Get the decompressed file
+            string fileIn = Path.GetDirectoryName(temp) + Path.DirectorySeparatorChar + "de" + Path.DirectorySeparatorChar + Path.GetFileName(temp);
+            Directory.CreateDirectory(Path.GetDirectoryName(fileIn));
+            DSDecmp.Main.Decompress(temp, fileIn, DSDecmp.Main.Get_Format(temp));

```

Comment:
To use this method you need to add a reference to the library in the project. That was the reason why the CI failed.